### ISSUE TITLE
byacc versus bison

### ISSUE THREAD

**Issue by fcorneli** [collectd/collectd#940]
When using byacc instead of bison, configure does not complain, but the build fails with:
```
Making all in liboconfig
make[3]: Entering directory `/home/user/collectd/src/liboconfig'
  YACC     parser.c
byacc: e - line 77 of "/home/user/collectd/src/liboconfig/parser.y", syntax error
%error-verbose
^
make[3]: *** [parser.c] Error 1
```

**Comment by mfournier** [comment]
@fcorneli could you give a bit more details on the context in which this would be a problem ? Did something in the docs made you believe byacc should favored over bison or something ? Thanks !

**Comment by fcorneli** [comment]
I just tried to build collectd on a system with byacc installed instead of bison and got this error message.

**Comment by ajayrockrock** [comment]
I was trying to build RPMs for my RHEL6 boxes and ran into this.  I have both byacc and bison installed:

    $ rpm -qa | grep byacc
    byacc-1.9.20070509-7.el6.x86_64

    $ rpm -qa | grep bison
    bison-runtime-2.4.1-5.el6.x86_64
    bison-2.4.1-5.el6.x86_64
    bison-devel-2.4.1-5.el6.x86_64

I'm not sure why it choose byacc over bison and stopped digging into the problem at this point.  :(

**Comment by jtyr** [comment]
When I was compiling CollectD for CentOS 7, I had to install the following packages to get it compiled (./build.sh && ./configure && make):
- bison
- byacc
- flex
- libedit
- libtool
- libtool-ltdl-devel

**Comment by octo** [comment]
*liboconfig*, collectd's config parsing, uses one or two features specific to *bison*, iIrc reporting of line numbers on which parse errors occurred.

We made this decision back then, because bison/byacc is only required when building from the source code repository, so having such a GNUism in there seemed acceptable.

**Comment by aferrari-technisys** [comment]
Hi @ajayrockrock @octo 

I has the same issue to build into Centos6 some workaround to this issue? 

Thanks!
Alejandro

**Comment by rubenk** [comment]
@aferrari-technisys yes, don't install byacc.

**Comment by aferrari-technisys** [comment]
@rubenk but I I don't has installed byacc build fail

# ./build.sh
WARNING: `yacc' not found!
    Please make sure that `yacc' is installed and is in one of the
    directories listed in the PATH environment variable.

I'm doing something wrong? some other way to build the RPM from Source? I'm testing a patch for write_tsdb for this reason I'm trying to build the RPM and distribute to all my nodes.

thanks!

**Comment by rubenk** [comment]
Do you have bison installed? Can you share your config.log?

**Comment by rubenk** [comment]
Should be fixed when #1465 is merged.

**Comment by rubenk** [comment]
Fixed by d23caf50d4cdd1ad41c8b373990f4de0b840c2d5
